developers/prerequisites/OSGi.md

```diff
@@ -0,0 +1,130 @@
+
+Services
+-------
+Another main concept, that allows the bundles to communicate between each other, is the *service* model. 
+
+**In OSGi, a bundle can register a *service* in a central [service registry](#important-definitions) under one ore more *service interface***. Published services also have service properties associated with them in the registry. It is an important feature of OSGi, because it provides a central place to register and get services. A bundle is permitted to register service objects at any time during the STARTING, ACTIVE or STOPPING states. Other bundles can go the registry and list all objects, that are registered under a specific interface or class. 
+
+A bundle can therefore register a service, it can get a service and it can track for appearing and disappearing of service. **Any number of bundles can register the same service type and any number of bundles can get the same service.** A simple diagram of the service usage and tracking is shown on Fig. 3.
+
+![OSGi Services][fig3]  
+Fig.3 OSGi Services (Source: <https://www.osgi.org/wp-content/uploads/services.png>)
+
+Important Definitions
+---------------------
+
+**bundle** - a unit of modularization, defined by the OSGi framework. A bundle is comprised of Java classes and other resources, which together can provide functions to end users. For more detailed definition - [OSGi Service Platform Core Specification, Chapter 3.2][OSGi-Core]
+
+**service** - any object that is registered in the OSGi Service Registry and can be looked up using its interface name(s). Definition - [OSGi Service Platform Core Specification, Chapter 5.1][OSGi-Core]
+
+**manifest** - descriptive information about the bundle, contained in its JAR file
+
+**Service Registry** - enables a bundle to publish objects to a shared registry, advertised via a given set of Java interfaces. 
+
+Further Reading
+----------
+ - [OSGi Service Platform Core Specification, Version 4.2][OSGi-Core]  
+ - [OSGi For Eclipse Developers - only the presentation materials](http://www.slideshare.net/caniszczyk/osgi-for-eclipse-developers-1331901)  
+ - [OSGi For Eclipse Developers - video](https://www.youtube.com/watch?v=4YfAo9ZoEGQ)  
+ - [OSGi API](https://osgi.org/javadoc/r4v42/index.html)  
+ - [OSGi Vogella guide](http://www.vogella.com/tutorials/OSGi/article.html)  
+ - [OSGi Framework Architecture – Three Conceptual Layers ](http://www.programcreek.com/2011/07/osgi-framework-architecture-three-conceptual-layers/)  
+ - [OSGi in action, Creating Modular Applications in Java](http://www.opus-college.net/devcorner/OSGi_in_Action.pdf)  
+ - [Lifecycle of a bundle](https://developer.atlassian.com/docs/atlassian-platform-common-components/plugin-framework/behind-the-scenes-in-the-plugin-framework/lifecycle-of-a-bundle)  
+ - <https://www.osgi.org/developer/where-to-start/> - this link contains references to books about OSGi
```

Comment:
I would remove "this link contains references to books about OSGi", since all the other links don't have a description either.